<launch>
  <arg name="nodelet_manager" default="nodelet_manager"/>
  <arg name="simulation" default="false"/>
  <arg name="robot_ip" default="192.168.3.112"/>

  <!-- Starting the nodelet-manager for fast data-transfer between nodelets -->
  <node ns="manager" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen">
    <param name="num_worker_threads" value="16"/>
  </node>

  <!-- Starting panda_simulation -->
  <node if="$(arg simulation)" name="read_pcd" pkg="pcl_filter" type="read_pcd" />

  <!-- Starting the realsense2-camera using the realsense2-launch-file -->
  <include unless="$(arg simulation)" file="$(find realsense2_camera)/launch/rs_camera.launch">
    <!-- Set the nodelet manager -->
    <arg name="external_manager" value="true"/>
    <arg name="manager" value="/manager/$(arg nodelet_manager)"/>

    <!-- Enable the Pointcloud (RGB-Image disabled) and other filters -->
    <!-- Filter settings can be adjusted online when running "rosrun rqt_reconfigure rqt_reconfigure"-->
    <arg name="filters" value="pointcloud,decimation,temporal"/>

    <!-- Point Cloud settings -->
    <arg name="pointcloud_texture_stream" value="RS2_STREAM_ANY"/>
    <!--  <arg name="enable_color" value="false" />  -->

    <!-- to adjust the frames per second these all three args need to uncommand -->
    <!--arg name="depth_fps" value="10"/-->
    <!--arg name="infra_fps" value="10"/-->
    <!--arg name="color_fps" value="10"/-->

    <!-- due to a bug with ocotmaps nodelets the input for this nodelet cannot be remaped   -->
    <!-- therefore the published topic must be in this odd namespace and with this odd name -->

  </include>

  <!-- load the settings for the Intel RealSense decimation filter of the camera -->
  <!-- settings can be adjusted online when running "rosrun rqt_reconfigure rqt_reconfigure" -->
  <node
      unless="$(arg simulation)"
      name="decimationSettingsLoader"
      pkg="dynamic_reconfigure"
      type="dynparam"
      args="load /camera/decimation $(find pcl_filter)/config/decimationFilterSettings.yaml"
      />

  <!-- load the settings for the Intel RealSense temporal filter of the camera -->
  <!-- settings can be adjusted online when running "rosrun rqt_reconfigure rqt_reconfigure" -->
  <node
      unless="$(arg simulation)"
      name="temporalSettingsLoader"
      pkg="dynamic_reconfigure"
      type="dynparam"
      args="load /camera/temporal $(find pcl_filter)/config/temporalFilterSettings.yaml"
      />

  
  <group ns="PCLNodelets">
    <!-- Starting PCL-Filter-Nodelet -->
    <node unless="$(arg simulation)" pkg="nodelet" type="nodelet" name="PCLFilterNodelet" args="load pcl_filter/PCLFilterNodelet /manager/$(arg nodelet_manager)"
          output="screen">
      <!-- due to a bug with ocotmaps nodelets the input for this nodelet cannot be remaped   -->
      <!-- therefore the published topic must be in this odd namespace and with this odd name -->
      <!-- <remap if="$(arg simulation)" from="/camera/depth/color/points" to="/pcl_output"/> -->
      <!-- <param if="$(arg simulation)" name="useDownsample" type="bool" value="true" /> -->
    </node>

    <!-- starting PCL-Cloud-Mirror-Nodelet -->
    <!-- <node pkg="nodelet" type="nodelet" name="PCLCloudMirrorNodelet" args="load pcl_filter/PCLCloudMirrorNodelet /manager/$(arg nodelet_manager)"
          output="screen">
    </node> -->
  </group>

  <!-- 在rviz中显示-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find pcl_filter)/config/seminar.rviz" required="true" />

</launch>
